README-rsync.md

```diff
@@ -0,0 +1,235 @@
+## OCluster/OBuilder macOS Infrastructure
+
+This document tells the ongoing story of supporting macOS in OCaml's Continuous Integration (CI) tools. In particular, how the OBuilder port is currently implemented, how this fits into the OCluster worker, and the manual process or keeping things up to date.
+
+<!-- TOC -->
+
+- [OCluster/OBuilder macOS Infrastructure](#oclusterobuilder-macos-infrastructure)
+- [The Big Picture](#the-big-picture)
+- [The Implementation](#the-implementation)
+    - [Overview](#overview)
+    - [Snapshotting Filesystem](#snapshotting-filesystem)
+    - [Global System Dependencies](#global-system-dependencies)
+    - [Tricking Homebrew with FUSE](#tricking-homebrew-with-fuse)
+    - [Rsyncing to Home](#rsyncing-to-home)
+    - [Docker-esque Base Images](#docker-esque-base-images)
+- [OCluster macOS Worker](#ocluster-macos-worker)
+    - [Building a macOS Base Directory](#building-a-macos-base-directory)
+- [Ansible Playbook](#ansible-playbook)
+- [Starting and Stopping](#start-stop)
+- [Current Deployment and Future Steps](#current-deployment-and-future-steps)
+- [Thanks](#thanks)
+
+<!-- /TOC -->
+
+## The Big Picture
+
+[OCurrent](https://github.com/ocurrent/ocurrent) is an OCaml library for building incremental pipelines. Pipelines can be thought of as directed graphs with each node being some kind of job that feeds its results into the next node. For an example, have a look at [the OCaml deploy pipeline](https://deploy.ci.ocaml.org/).
+
+An integral part to most of the [OCaml pipelines](https://github.com/ocurrent/overview) is something called [OCluster](https://github.com/ocurrent/ocluster) and [OBuilder](https://github.com/ocurrent/obuilder). OCluster is a library and collection of binaries for managing an infrastructure for building and scheduling jobs. These jobs could be Docker jobs or OBuilder jobs. OBuilder is quite similar to `docker build` but written in OCaml and using just a notion of *an execution environment* (on Linux [`runc`](https://github.com/opencontainers/runc)) and a *snapshotting filesystem* ([Btrfs](https://btrfs.wiki.kernel.org/index.php/Main_Page), [ZFS](https://openzfs.org/wiki/Main_Page), or an inefficient but convenient copying backend using `rsync`).
```

Comment:
Should `runc` be in monospace or like this: runc ?
Same for `rsync`. I've seen it written Rsync when short for Remote Sync and rsync in Wikipedia. Or is it a command `rsync`. It seems throughout this document it's used sometimes as a command and sometime as the utility itself.

Reply:
`runc` and `rsync` are both system commands.
* https://github.com/opencontainers/runc
* https://rsync.samba.org or https://github.com/WayneD/rsync